test/plugin/base_test.rb

```diff
@@ -946,26 +946,42 @@ def test_split_logs_by_tag
   def test_timestamps
     setup_gce_metadata_stubs
     current_time = Time.now
+    one_day_later = current_time.to_datetime.next_day.to_time
+    just_under_one_day_later = one_day_later - 1
     next_year = Time.mktime(current_time.year + 1)
     one_second_before_next_year = next_year - 1
-    adjusted_to_last_year =
-      one_second_before_next_year.to_datetime.prev_year.to_time
     one_second_into_next_year = next_year + 1
-    one_day_into_next_year = next_year.to_date.next_day.to_time
+    one_day_into_next_year = next_year.to_datetime.next_day.to_time
+    # Note: these tests behave differently on December 31.
+    # TODO: use patched Time.now instead of messing with the current time.
```

Comment:
Suggestion: Instead of patching Time.now, how about making "now" a parameter to compute_timestamp so that it's a pure function, then testing that function directly?

I feel like compute_timestamp is complex enough that it deserves its own unit test.

Reply:
Great idea. Done.